yt_dlp/extractor/beeg.py

```diff
@@ -36,81 +48,49 @@ class BeegIE(InfoExtractor):
         # api/v6 v2 w/o t
         'url': 'https://beeg.com/1277207756',
         'only_matching': True,
-    }, {
-        'url': 'https://beeg.porn/video/5416503',
-        'only_matching': True,
-    }, {
-        'url': 'https://beeg.porn/5416503',
-        'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
 
-        beeg_version = self._search_regex(
-            r'beeg_version\s*=\s*([\da-zA-Z_-]+)', webpage, 'beeg version',
-            default='1546225636701')
+        video = self._download_json(
+            'https://store.externulls.com/facts/file/%s' % video_id,
+            video_id, 'Downloading JSON for %s' % video_id)
 
-        if len(video_id) >= 10:
-            query = {
-                'v': 2,
-            }
-            qs = parse_qs(url)
-            t = qs.get('t', [''])[0].split('-')
-            if len(t) > 1:
-                query.update({
-                    's': t[0],
-                    'e': t[1],
-                })
-        else:
-            query = {'v': 1}
+        fc_facts = video.get('fc_facts')
+        firstKey = None
+        for i in range(len(fc_facts)):
+            if not firstKey or fc_facts[i - 1]['id'] < fc_facts[firstKey]['id']:
+                firstKey = i - 1
```

Comment:
I didnt test,  but pretty sure this is wrong

Reply:
Yea, there was 2 problems, 
i forgot it starts with 0, and i forgot list keys is not the same for multiple iterations on same code. changed the code to save first fact not first key.